tidy3d/components/geometry/primitives.py

```diff
@@ -215,6 +225,123 @@ def _only_middle_for_infinite_length_slanted_cylinder(cls, val, values):
             )
         return val
 
+    def to_polyslab(
+        self, num_pts_circumference: int = _N_PTS_CYLINDER_POLYSLAB, **kwargs
+    ) -> PolySlab:
+        """Convert instance of ``Cylinder`` into a discretized version using ``PolySlab``.
+
+        Parameters
+        ----------
+        num_pts_circumference : int = 51
+            Number of points in the circumference of the discretized polyslab.
+        **kwargs:
+            Extra keyword arguments passed to ``PolySlab()``, such as ``sidewall_angle``.
+
+        Returns
+        -------
+        PolySlab
+            Extruded polygon representing a discretized version of the cylinder.
+        """
+
+        center_axis = self.center_axis
+        length_axis = self.length_axis
+        slab_bounds = (center_axis - length_axis / 2.0, center_axis + length_axis / 2.0)
+
+        if num_pts_circumference < 3:
+            raise ValueError("'PolySlab' from 'Cylinder' must have 3 or more radius points.")
+
+        _, (x0, y0) = self.pop_axis(self.center, axis=self.axis)
+
+        xs_, ys_ = self._points_unit_circle(num_pts_circumference=num_pts_circumference)
+
+        xs = x0 + self.radius * xs_
+        ys = y0 + self.radius * ys_
+
+        vertices = anp.stack((xs, ys), axis=-1)
+
+        return PolySlab(
+            vertices=vertices,
+            axis=self.axis,
+            slab_bounds=slab_bounds,
+            **kwargs,
+        )
+
+    def _points_unit_circle(
+        self, num_pts_circumference: int = _N_PTS_CYLINDER_POLYSLAB
+    ) -> np.ndarray:
+        """Set of x and y points for the unit circle when discretizing cylinder as a polyslab."""
+        angles = np.linspace(0, 1, num_pts_circumference + 1)[:-1]
+        xs = np.cos(angles)
+        ys = np.sin(angles)
+        return np.stack((xs, ys), axis=0)
+
+    def compute_derivatives(self, derivative_info: DerivativeInfo) -> AutogradFieldMap:
+        """Compute the adjoint derivatives for this object."""
+
+        # compute number of points in the diameter of the polyslab using resolution info
+        wvl0 = C_0 / derivative_info.frequency
+        wvl_mat = wvl0 / max(1.0, np.sqrt(abs(derivative_info.eps_in)))
+
+        diameter = 2 * np.pi * self.radius
+        wvls_in_diameter = diameter / wvl_mat
```

Comment:
Again it seems like you mean circumference and not diameter... diameter is just 2 * radius. So the value is correct since you're computing `num_pts_circumference` but the names should be `circumference `and `wvls_in_circumference`?

Reply:
oops, just missed these variables. fixed